problem_builder/tests/unit/test_instructor_tool.py

```diff
@@ -0,0 +1,226 @@
+        block.runtime = self.runtime_mock
+        block.children = []
+        for attr, val in block_info.get('attrs', {}).items():
+            setattr(block, attr, val)
+        return block
+
+    def setUp(self):
+        self.runtime_mock = Mock()
+        self.runtime_mock.get_block = self._get_block
+        scope_ids_mock = Mock()
+        scope_ids_mock.usage_id = u'0'
+        self.block = InstructorToolBlock(
+            self.runtime_mock, field_data=DictFieldData({}), scope_ids=scope_ids_mock
+        )
+        self.block.children = [
+            # No attributes: Prefer usage_id
+            {'usage_id': u'1'},
+            # Single attribute: Prefer attribute that's present
+            {'usage_id': u'2', 'preferred_attr': 'question', 'attrs': {'question': 'question'}},
+            {'usage_id': u'3', 'preferred_attr': 'name', 'attrs': {'name': 'name'}},
+            {'usage_id': u'4', 'preferred_attr': 'display_name', 'attrs': {'display_name': 'display_name'}},
+            # Two attributes (question, name): Prefer question
+            {
+                'usage_id': u'5',
+                'preferred_attr':
+                'question', 'attrs': {'question': 'question', 'name': 'name'}
+            },
+            # Two attributes (question, display_name): Prefer question
+            {
+                'usage_id': u'6',
+                'preferred_attr': 'question',
+                'attrs': {'question': 'question', 'display_name': 'display_name'}
+            },
+            # Two attributes (name, display_name): Prefer name
+            {
+                'usage_id': u'7',
+                'preferred_attr': 'name',
+                'attrs': {'name': 'name', 'display_name': 'display_name'}
+            },
+            # All attributes: Prefer question
+            {
+                'usage_id': u'8',
+                'preferred_attr': 'question',
+                'attrs': {'question': 'question', 'name': 'name', 'display_name': 'display_name'}
+            },
+        ]
+
+    def test_student_view_block_tree_names(self):
+        """
+        Check if `student_view` method uses preferred block attributes
+        when building block tree for render context.
+
+        Each entry of a block tree is a dictionary that must contain
+        an `id` key and a `name` key.
+
+        - `id` must be set to the ID
+          (usage_id or block_id) of the corresponding block.
+
+        - `name` must be set to the value of one of the following
+          attributes of the corresponding block:
+
+          - question
+          - name (question ID)
+          - display_name (question title)
+          - block ID
+
+          Note that the attributes are listed in order of preference;
+          i.e., if `block.question` has a meaningful value, that value
+          should be used for `name` (irrespective of what the values
+          of the other attributes might be).
+        """
+        instructor_tool.loader = Mock()
```

Comment:
@itsjeyd looks like you're monkey-patching `loader` in instructor_tool. I believe the correct way for this would be to use mock.patch as context manager:

    with mock.patch('instructor_tool.loader`) as patched_loader:
        ...